### PR TITLE
libbpf-tools: use raw_tp sched_switch instead of kprobe

### PR DESCRIPTION
**Pull request by ethercflow**
To fix https://github.com/iovisor/bcc/issues/3293 metioned.

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### CODE REVIEW

File: libbpf-tools/cpudist.bpf.c [hunk 1]
```diff
@@ -75,18 +75,18 @@ static __always_inline void update_hist(struct task_struct *task,
...
-	u32 tgid = id >> 32, pid = id;
+	u32 tgid = next->tgid, pid = next->pid;
 	u64 ts = bpf_ktime_get_ns();
 
 	if (targ_offcpu) {
 		store_start(prev_tgid, prev_pid, ts);
-		update_hist((void*)bpf_get_current_task(), tgid, pid, ts);
+		update_hist(next, tgid, pid, ts);
 	} else {
 		if (BPF_CORE_READ(prev, state) == TASK_RUNNING)
```
Comment (anakryiko) [inline libbpf-tools/cpudist.bpf.c:91]: why not get rid of BPF_CORE_READs everywhere, given we use tp_btf?

Comment (ethercflow) [reply libbpf-tools/cpudist.bpf.c]: Oh I missed them, only pay attention to the longest `BPF_CORE_READ_STR_INTO`. (:   Fixed, thx.

File: libbpf-tools/cpudist.bpf.c [hunk 2]
```diff
@@ -62,7 +62,7 @@ static __always_inline void update_hist(struct task_struct *task,
 		histp = bpf_map_lookup_elem(&hists, &id);
 		if (!histp)
 			return;
-		BPF_CORE_READ_STR_INTO(&histp->comm, task, comm);
+		bpf_probe_read_str(&histp->comm, sizeof(task->comm), task->comm);
```
Comment (anakryiko) [inline libbpf-tools/cpudist.bpf.c:65]: missed this, needs to be bpf_probe_read_kernel_str()

**Review by anakryiko — APPROVED** [review]
please fix probe_read_str and otherwise looks good

**Review by anakryiko — APPROVED** [review]
Looks great, thanks!

**Comment by ethercflow** [comment]
> Looks great, thanks!

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Pull request merged**